src/prefect/cli/server.py

```diff
@@ -153,11 +184,32 @@ async def start(
             # https://github.com/PrefectHQ/server/issues/2475
 
             setup_signal_handlers_server(
-                server_process_id, "the Prefect server", app.console.print
+                process_id, "the Prefect server", app.console.print
             )
-    except CancelledError:
-        logger.debug("Server task group cancelled")
 
+            await consume_process_output(process, sys.stdout, sys.stderr)
+
+    except anyio.EndOfStream:
+        logging.error("Subprocess stream ended unexpectedly")
+    except Exception as e:
+        logging.error(f"An error occurred: {str(e)}")
+
+    app.console.print("Server stopped!")
+
+
+@server_app.command()
+async def stop():
+    """Stop a Prefect server instance running in the background"""
+    pid_file = anyio.Path(PREFECT_HOME.value() / "server.pid")
+    if not await pid_file.exists():
+        exit_with_error("No server running in the background.")
+    pid = int(await pid_file.read_text())
+    try:
+        os.kill(pid, 15)
+    except ProcessLookupError:
+        exit_with_error("No server running in the background.")
```

Comment:
I wouldn't consider this case an error.  They wanted the process to not be running, and it's not running.  I'd make this a successful exit (even if you still print the message that no server was running), which should make it easier to use in automated scripts.  (I _would_ keep the message on L205 an error exit code)